services/catalog/src/simcore_service_catalog/services/catalog.py

```diff
@@ -41,41 +90,157 @@ async def list_services_paginated(
         ((s.key, s.version) for s in services_in_db), product_name=product_name
     )
+    if not services_access_rights:
+        raise CatalogForbiddenError(
+            name="any service",
+            user_id=user_id,
+            product_name=product_name,
+        )
 
     # NOTE: aggregates published (i.e. not editable) is still missing in this version
-
     items = [
-        ServiceGetV2(
-            key=db.key,
-            version=db.version,
-            name=db.name,
-            thumbnail=db.thumbnail or None,
-            description=db.description,
-            version_display=f"V{db.version}",  # rg.version_display,
-            type=_deduce_service_type_from(db.key),  # rg.service_type,
-            badges=[
-                Badge.Config.schema_extra["example"],
-            ],  # rg.badges,
-            contact=Author.Config.schema_extra["examples"][0]["email"],  # rg.contact,
-            authors=Author.Config.schema_extra["examples"],
-            owner=db.owner_email or None,
-            inputs={},  # rg.inputs,
-            outputs={},  # rg.outputs,
-            boot_options=None,  # rg.boot_options,
-            min_visible_inputs=None,  # rg.min_visible_inputs,
-            access_rights={
-                a.gid: ServiceGroupAccessRightsV2.construct(
-                    execute=a.execute_access,
-                    write=a.write_access,
-                )
-                for a in db_ar
-            },  # db.access_rights,
-            classifiers=db.classifiers,
-            quality=db.quality,
-            history=[h.to_api_model() for h in db.history],
-        )
+        _to_api_model(db, db_ar)
         for db in services_in_db
         if (db_ar := services_access_rights.get((db.key, db.version)))
     ]
 
     return total_count, items
+
+
+async def get_service(
+    repo: ServicesRepository,
+    # image_registry,
+    product_name: ProductName,
+    user_id: UserID,
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
+) -> ServiceGetV2:
+
+    db_ar = await repo.get_service_access_rights(
+        key=service_key,
+        version=service_version,
+        product_name=product_name,
+    )
+    if not db_ar:
+        raise CatalogItemNotFoundError(
+            name=f"{service_key}:{service_version}",
+            service_key=service_key,
+            service_version=service_version,
+            user_id=user_id,
+            product_name=product_name,
+        )
+
+    db = await repo.get_service_with_history(
+        product_name=product_name,
+        user_id=user_id,
+        key=service_key,
+        version=service_version,
+    )
+    if not db:
+        raise CatalogForbiddenError(
+            name=f"{service_key}:{service_version}",
+            service_key=service_key,
+            service_version=service_version,
+            user_id=user_id,
+            product_name=product_name,
+        )
+
+    return _to_api_model(db, db_ar)
```

Comment:
we are back to 8-bit variables? ;)
since you are in get_service, I guess that access_rights and service are ok to use ;)

Reply:
When the context is clear, I find it easier to read. 
db: identifies the database
db_ar: identifies the access-rights part of the database

I will make them "16 bit" nonetheless :-)